go-controller/pkg/cni/helper_linux.go

```diff
@@ -817,3 +831,71 @@ func (pr *PodRequest) deletePorts(ifaces []string, podNamespace, podName string)
 		pr.deletePort(iface, podNamespace, podName)
 	}
 }
+
+func getLinkLocalIP(namespace, podName, netName string, getter PodAndNodeInfoGetter) (string, error) {
+	pod, err := getter.getPod(namespace, podName)
+	if err != nil {
+		return "", err
+	}
+
+	node, err := getter.getNode(pod.Spec.NodeName)
+	if err != nil {
+		return "", err
+	}
+
+	lrpAddress, err := util.ParseNodeGatewayRouterJoinNetwork(node, netName)
+	if err != nil {
+		return "", err
+	}
+
+	if lrpAddress.IPv6 == "" {
+		return "", nil
+	}
+
+	// TODO: Support ipv6 single stack (mac is generated with ipv6)
+	// LRP's mac is calcualted from the join subnet IPv4
+	lrpAddressIPv4, _, err := net.ParseCIDR(lrpAddress.IPv4)
+	if err != nil {
+		return "", err
+	}
+	raSrcMAC := util.IPAddrToHWAddr(lrpAddressIPv4)
+	return util.HWAddrToIPv6LLA(raSrcMAC).String(), nil
+}
+
+func setupIngressFilter(linkLocalIP string) error {
+	tableName := "ingress_filter"
+	chainName := "input"
+	deviceName := "ovn-udn1"
+
+	nft, err := knftables.New(knftables.NetDevFamily, tableName)
+	if err != nil {
+		return fmt.Errorf("failed to initialize table: %w", err)
+	}
+
+	tx := nft.NewTransaction()
+
+	tx.Add(&knftables.Table{})
+
+	tx.Add(&knftables.Chain{
+		Name:     chainName,
+		Type:     knftables.PtrTo(knftables.FilterType),
+		Hook:     knftables.PtrTo(knftables.IngressHook),
+		Priority: knftables.PtrTo(knftables.FilterPriority),
+		Device:   knftables.PtrTo(deviceName),
+	})
+
+	tx.Add(&knftables.Rule{
+		Chain: chainName,
+		Rule: knftables.Concat(
+			"icmpv6", "type", "nd-router-advert", "ip6", "saddr", "!=", linkLocalIP, "@th,48,16", "!=", "0", "drop",
+		),
+	})
+
+	err = nft.Run(context.TODO(), tx)
```

Comment:
```suggestion
	err = nft.Run(context.Background(), tx)
```

Reply:
np will change, but note that all over we are using nft.Run with context.TODO

Reply:
it's a nit, it's pretty much the same thing. 

Still, TODO means you mean to change it in the future for something else. Do you ? 
And FWIW, I actually think we should put a context w/ a timeout of a couple of seconds. Let's say 5. Arbitrary number.

Thoughts ?

Reply:
I don't think we should add timeout, while all the others don't have, and since we don't know about any problems with it as it is now.
Just if we will see any of those is doing problems we can add